### DISCUSSION TITLE
Running the tutorials in MOOSE

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#26207] [Q&A Getting Started]
I have some questions.

1. After creating my application on MOOSE how do I run the test examples in the MOOSE repository in my application? I followed the MOOSE workshop 2023 and I made the logan1.i  file and some other changes but it seems I was not committing the new additions and I got a lot of issues with my MOOSE. 

2.  This is a follow-up question can i copy let's say the step01 directory into my application and create the logan1.i file so it does not affect my MOOSE in general?

**Comment by GiudGiud** [comment]
1) You need to compile an executable then use it on the input files.
Please elaborate on the issues if you need help there

2) yes that s not going to affect MOOSE. Moving input files will not break the code

**Reply by Emmanuel-324** [reply]
(python-stack) [emmanuel324@tinkercliffs1 step01_diffusion]$ make -j4
Makefile:20: /home/emmanuel324/projects/belson/logan1/../..//framework/build.mk: No such file or directory
Makefile:21: /home/emmanuel324/projects/belson/logan1/../..//framework/moose.mk: No such file or directory
Makefile:41: /home/emmanuel324/projects/belson/logan1/../..//modules/modules.mk: No such file or directory
Makefile:49: /home/emmanuel324/projects/belson/logan1/../..//framework/app.mk: No such file or directory
make: *** No rule to make target `/home/emmanuel324/projects/belson/logan1/../..//framework/app.mk'.  Stop.

So from the snippet above, I copied the step01_diffusion folder into my application and I was trying to compile an executable by running make -j4. That's what I did in the original step01_diffusion in the MOOSE repository and it creates  the darcy_thermo_mech-opt

**Reply by GiudGiud** [reply]
there needs to be a makefile to be able to use make

**Reply by Emmanuel-324** [reply]
Yes, there is a makefile in the step01_diffusion folder I copied to my application.

In it original directory in the MOOSE when I run make in the step01_diffusion folder it works but when I copied the same step01_diffusion folder into my application and use make it does not work as shown in the previous submission.

**Reply by GiudGiud** [reply]
oh ok. Can you set MOOSE_DIR to the moose folder that you are using to store moose?
The makefile will need it

**Reply by Emmanuel-324** [reply]
I do not think I understand what you mean.

**Reply by GiudGiud** [reply]
MOOSE_DIR is an environment variable in your system.
likely you should do:
```
export MOOSE_DIR = ~/projects/your_app/moose
```

**Reply by Emmanuel-324** [reply]
(python-stack) [emmanuel324@tinkercliffs1 moose]$ export MOOSE_DIR = ~/projects/belson/moose
-bash: export: `=': not a valid identifier
-bash: export: `/home/emmanuel324/projects/belson/moose': not a valid identifier

**Reply by GiudGiud** [reply]
```
export MOOSE_DIR="~/projects/your_app/moose"
```
maybe?

this is linux stuff, you should google it on the side to make it work

**Reply by Emmanuel-324** [reply]
I am on an HPC cluster.

**Reply by Emmanuel-324** [reply]
Okay. If I want to use the original MOOSE directory, what is the best and right way to commit any new input files I create in there or any new changes to prevent the warning of committing the changes?

**Reply by GiudGiud** [reply]
If you want to use a fork of the MOOSE repo to do your development, that's not the preferred way but it's workable.

You can commit new inputs or modifications to existing inputs to your fork.
what do you mean by `the warning of committing the changes` ?
